lib/manageiq/cross_repo/runner.rb

```diff
@@ -35,14 +37,22 @@ def run
 
     def run_tests
       with_test_env do
-        system!({"TRAVIS_BUILD_DIR" => test_repo.path.to_s}, "bash", "tools/ci/before_install.sh") if ENV["CI"] && File.exist?("tools/ci/before_install.sh")
-        system!(env_vars, "bin/setup")
-        system!(script_cmd)
+        require "yaml"
+        travis_yml = YAML.load_file(".travis.yml")
+
+        commands =
+          Array(travis_yml["before_install"]) +
+          Array(travis_yml["install"] || install_cmd) +
+          Array(travis_yml["before_script"]) +
+          Array(travis_yml["script"] || script_cmd) +
+          Array(travis_yml["after_script"])
```

Comment:
:+1: Array() will handle both styles that travis.yml supports:
```
script: cmd arg

script:
- cmd arg
```

Reply:
And it also handles `nil` when the .yml misses some sections. Neat :)  Worth a comment if you don't have a test.

Reply:
:+1: I think this deserves a spec test and a comment